src/components/views/voip/CallLobby.tsx

```diff
@@ -147,18 +129,36 @@ export const CallLobby: FC<Props> = ({ room, call }) => {
         setVideoMuted(!videoMuted);
     }, [videoMuted, setVideoMuted]);
 
-    const videoStream = useAsyncMemo(async () => {
+    const [videoStream, audioInputs, videoInputs] = useAsyncMemo(async () => {
         if (videoInputId && !videoMuted) {
             try {
-                return await navigator.mediaDevices.getUserMedia({
+                // We get the preview stream before requesting devices: this is because
+                // we need (in some browsers) an active media stream in order to get
+                // non-blank labels for the devices. According to the docs, we
+                // need a stream of each type (audio + video) if we want to enumerate
+                // audio & video devices, although this didn't seem to be the case
+                // in practice for me. We request both anyway.
+                const s = await navigator.mediaDevices.getUserMedia({
                     video: { deviceId: videoInputId },
+                    audio: { deviceId: audioInputId },
                 });
+
+                const devices = await MediaDeviceHandler.getDevices();
+                return [s, devices[MediaDeviceKindEnum.AudioInput], devices[MediaDeviceKindEnum.VideoInput]];
             } catch (e) {
                 logger.error(`Failed to get stream for device ${videoInputId}`, e);
             }
         }
         return null;
```

Comment:
This will cause the destructuring to fail and throw an error. I also think I would expect the devices to still be returned in this case, somehow

Reply:
Oh yep, good spot. Not sure what the best solution is here so have just always requested a stream, as per comments. Open to better ideas though.